pkg/usagetracker/tracker_store_snapshot.go

```diff
@@ -44,6 +46,32 @@ func (t *trackerStore) snapshot(shard uint8, now time.Time, buf []byte) []byte {
 	return snapshot.Get()
 }
 
+// loadSnapshots loads the snapshots from the given shards concurrently with GOMAXPROCS workers.
+// This speeds up the snapshot loading process as each shard can be loaded independently.
+// This reduces the amount of time track requests spend waiting on shard locks.
+func (t *trackerStore) loadSnapshots(shards [][]byte, now time.Time) error {
+	jobs := make(chan []byte, len(shards))
+	for _, shard := range shards {
+		jobs <- shard
+	}
+	close(jobs)
+
+	workers := runtime.GOMAXPROCS(0)
```

Comment:
`runtime.GOMAXPROCS(0)` may be a lot in a large shared cluster, right? Is it smaller or bigger than number of shards?